src/doc/trpl/unsafe.md

```diff
@@ -649,53 +649,146 @@ it exists. The marker is the attribute `#[lang="..."]` and there are
 various different values of `...`, i.e. various different 'lang
 items'.
 
-For example, `Box` pointers require two lang items, one for allocation
-and one for deallocation. A freestanding program that uses the `Box`
-sugar for dynamic allocations via `malloc` and `free`:
+For example, there are lang items related to the implementation of
+string slices (`&str`); one of these is `str_eq`, which implements the
```

Comment:
We already talk about this all the time, so you can just say 'string slices' or `&str`, rather than both. I guess it doesn't hurt though.